recipes/fmt/all/conanfile.py

```diff
@@ -97,37 +100,31 @@ def package(self):
             rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "fmt"
-        self.cpp_info.names["cmake_find_package_multi"] = "fmt"
-        self.cpp_info.names["pkg_config"] = "fmt"
-
         target = "fmt-header-only" if self.options.header_only else "fmt"
-        self.cpp_info.set_property("cmake_target_name", "fmt::{}".format(target))
+        self.cpp_info.set_property("cmake_file_name", "fmt")
+        self.cpp_info.set_property(f"cmake_target_name", f"fmt::{target}")
+        self.cpp_info.set_property("pkg_config_name",  "fmt")
 
         # TODO: back to global scope in conan v2 once cmake_find_package* generators removed
-        self.cpp_info.components["_fmt"].includedirs.extend(["include"])
+        if self.options.with_fmt_alias:
+            self.cpp_info.components["_fmt"].defines.append("FMT_STRING_ALIAS=1")
+
         if self.options.header_only:
             self.cpp_info.components["_fmt"].defines.append("FMT_HEADER_ONLY=1")
-            if self.options.with_fmt_alias:
-                self.cpp_info.components["_fmt"].defines.append("FMT_STRING_ALIAS=1")
         else:
             postfix = "d" if self.settings.build_type == "Debug" else ""
             libname = "fmt" + postfix
             self.cpp_info.components["_fmt"].libs = [libname]
             if self.settings.os == "Linux":
                 self.cpp_info.components["_fmt"].system_libs.extend(["m"])
-            # FIXME: remove when Conan 1.50 is used in c3i and update the Conan required version
-            # from that version components don't have empty libdirs by default
-            self.cpp_info.components["_fmt"].libdirs.extend(["lib"])
-            self.cpp_info.components["_fmt"].bindirs.extend(["bin"])
-            if self.options.with_fmt_alias:
-                self.cpp_info.components["_fmt"].defines.append("FMT_STRING_ALIAS=1")
             if self.options.shared:
                 self.cpp_info.components["_fmt"].defines.append("FMT_SHARED")
 
         # TODO: to remove in conan v2 once cmake_find_package* generators removed
+        self.cpp_info.names["cmake_find_package"] = "fmt"
+        self.cpp_info.names["cmake_find_package_multi"] = "fmt"
+        self.cpp_info.names["pkg_config"] = "fmt"
         self.cpp_info.components["_fmt"].names["cmake_find_package"] = target
         self.cpp_info.components["_fmt"].names["cmake_find_package_multi"] = target
-        self.cpp_info.components["_fmt"].set_property("cmake_target_name", "fmt::{}".format(target))
-        # FIXME: Remove as soon as Conan client provide a hotfix. See conan-io/conan-center-index#12149
-        self.cpp_info.components["_fmt"].builddirs = [""]
+        self.cpp_info.components["_fmt"].set_property("cmake_target_name", f"fmt::{target}")
+        self.cpp_info.components["_fmt"].includedirs.extend(["include"])
```

Comment:
why?

Reply:
It was already there 🙈 I am not sure what it's for...
I think because it was a super early v2 migration there was a workaround or fix maybe?

Reply:
I think it's unnecessary since 1.50.0 (same issue than lib & bin in components)